### PR TITLE
fast blockmap-based line-of-sight calculation

### PR DESCRIPTION
**Pull request by fabiangreffrath**

### CODE REVIEW

**Comment by rfomin** [comment]
Wow this is amazing, I got 2x more FPS in the beginning of the https://dsdarchive.com/wads/cosmogenesis

**Comment by fabiangreffrath** [comment]
> Wow this is amazing, I got 2x more FPS in the beginning of the https://dsdarchive.com/wads/cosmogenesis

Hm, I don't see much of a difference. Which place in which map do you mean?

**Comment by rfomin** [comment]
> Hm, I don't see much of a difference. Which place in which map do you mean?

Running this little demo with the `-timedemo` parameter: [test1.zip](https://github.com/fabiangreffrath/woof/files/14424172/test1.zip), I got about 50% more FPS.

**Comment by fabiangreffrath** [comment]
o_O Indeed! From 170 FPS to 224 FPS, that's a 30% win.

BTW, formerly the in-game FPS counter would update once per second. Now, it updates very fast and it is hard to tell an average. Could this get changed back, please?

**Comment by rfomin** [comment]
> BTW, formerly the in-game FPS counter would update once per second. Now, it updates very fast and it is hard to tell an average. Could this get changed back, please?

Done in https://github.com/fabiangreffrath/woof/commit/f8906440ebe207d3b8cdb3486f62b8fda1001606

File: src/m_menu.c [hunk 1]
```diff
@@ -3669,6 +3669,9 @@ setup_menu_t comp_settings1[] =  // Compatibility Settings screen #1
   {"Improved Hit Detection", S_ONOFF|S_STRICT|S_BOOM, m_null, M_X,
    M_SPC, {"blockmapfix"}},
 
+  {"Fast Line-of-Sight Calculation", S_ONOFF|S_STRICT, m_null, M_X,
```
Comment (fabiangreffrath) [inline src/m_menu.c:3672]: The wording is of course subject to discussion.

**Review by rfomin — APPROVED** [review]
Looks good, thanks!

Maybe we should write somewhere that it was taken from Heretic.

**Pull request merged**